doc/guides/device_mgmt/index.rst

```diff
@@ -39,14 +39,466 @@ functionality over BLE and serial.
+       The ``taskstat`` command itself can be enabled by setting
+       :option:`CONFIG_OS_MGMT_TASKSTAT`. :option:`CONFIG_THREAD_MONITOR` also
+       needs to be enabled otherwise a ``-8`` (``MGMT_ERR_ENOTSUP``) will be
+       returned.
+
+.. tip::
+
+    ``taskstat`` has a few options that might require tweaking. The
+    :option:`CONFIG_THREAD_NAME` must be set to display the task names, otherwise
+    the priority is displayed. Since the ``taskstat`` packets are large, they
+    might need increasing the :option:`CONFIG_MCUMGR_BUF_SIZE` option.
+
+.. warning::
+
+    To display the correct stack size in the ``taskstat`` command, the
+    :option:`CONFIG_THREAD_STACK_INFO` option must be set.
+    To display the correct stack usage in the ``taskstat`` command, both
+    :option:`CONFIG_THREAD_STACK_INFO` and :option:`CONFIG_INIT_STACKS` options
+    must be set.
+
+.. _image_mgmt:
+
+Image Management
+****************
+
+The image management provided by ``mcumgr`` is  based on the image format defined
+by MCUboot. For more details on the internals see `MCUboot design`_ and :ref:`west-sign`.
+
+To list available images in a device::
+
+  mcumgr <connection-options> image list
+
+This should result in an output similar to this::
+
+  $ mcumgr -c acm0 image list
+  Images:
+    image=0 slot=0
+      version: 1.0.0
+      bootable: true
+      flags: active confirmed
+      hash: 86dca73a3439112b310b5e033d811ec2df728d2264265f2046fced5a9ed00cc7
+  Split status: N/A (0)
+
+Where ``image`` is the number of the image pair in a multi-image system, and slot
+is the number of the slot where the image is stored, ``0`` for primary and ``1`` for
+secondary. This image being ``active`` and ``confirmed`` means it will run again on
+next reset. Also relevant is the ``hash``, which is used by other commands to
+refer to this specific image when performing operations.
+
+An image can be manually erased using::
+
+  mcumgr <connection-options> image erase
+
+The behavior of ``erase`` is defined by the server (``mcumgr`` in the device).
+The current implementation is limited to erasing the image in the secondary
+partition.
+
+To upload a new image::
+
+  mcumgr <connection-options> image upload [-n] [-e] [-u] <signed-bin>
+
+* ``-n``: This option allows uploading a new image to a specific set of images
+  in a multi-image system, and is currently only supported by MCUboot when the
+  ``BOOT_SERIAL`` option is enabled.
+
+* ``-e``: This option avoids performing a full erase of the partition before
+  starting a new upload.
+
+.. tip::
+
+   The ``-e`` option should always be passed in because the ``upload`` command
+   already checks if an erase is required, respecting the
+   :option:`CONFIG_IMG_ERASE_PROGRESSIVELY` setting.
+
+.. warning::
+
+   ``mcumgr`` does not understand .hex files, when uploading a new image always
+   use the .bin file.
+
+* ``-u``: upgrade only to newer image version.
```

Comment:
So does mcumgr-cli negotiate whether image can be uploat at all? How this works?

Reply:
Negotiate? This is just a CBOR flag that is passed to the `mcumgr` server, who checks that the uploaded image is newer than the one already present, here: https://github.com/apache/mynewt-mcumgr/blob/master/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c#L498. I don't remember if I actually tested this or just read through the code.

Reply:
understood